tests/models/soil/test_soil_model.py

```diff
@@ -6,7 +6,7 @@
 
 import numpy as np
 import pytest
-from dotmap import DotMap  # type: ignore
+from scipy.optimize import OptimizeResult  # type: ignore
```

Comment:
Why are you ignoring the type here?

Reply:
As far as I can tell `scipy.optimize` is not typed even though other `scipy` modules are typed. I could have missed them but also couldn't track down type stubs